dali/dfu/dfuutil.cpp

```diff
@@ -1296,6 +1296,24 @@ class CDFUhelper: implements IDFUhelper, public CInterface
         }
     }
 
+    MemoryBuffer &getFileHistory(const char *lfn,MemoryBuffer &out, IUserDescriptor *user)
+    {
+        Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(lfn, user);
+        if (!file) {
+            throwError1(DFUERR_DFileNotFound, lfn);
+        }
+        else
+        {
+            Owned<IPropertyTree> t = queryDistributedFileDirectory().getFileTree(lfn, user)->getPropTree("History");
```

Comment:
Also why use IPropertyTree, better for a few reasons to get IDistributedFile and use queryHistory()
One reason is, getFileTree will serialize the whole of the file meta, lookup() won't.